app/core/auth/__init__.py

```diff
@@ -10,40 +11,64 @@ def admin_auth_check_decorator(func):
+
+        if user_data.user_metadata.get('user_type') != 'admin':
+            raise PermissionException("Unauthorized access. User is not admin.")
+
         return await func(*args, **kwargs)
+
     return wrapper
 
 
 def chatbot_auth_check_decorator(func):
-    '''checks a predefined token in request header'''
+    '''checks a predefined token in request header, and returns available sources.
+    Note: We will probably want to split this into two functions: one for auth check 
+    and one to look up the sources.
+    '''
     @wraps(func)
     async def wrapper(*args, **kwargs):
-        # To be implemented.
-        # Uses Supabase, gets user roles
-        # Verify the DB requesting to connect to and sources listed for querying are permissible
-        # Raises 403 error for unauthorized access
-        # Passes the list of accessible sources to the calling function
-
-        # url: str = os.environ.get("SUPABASE_URL")
-        # key: str = os.environ.get("SUPABASE_KEY")
-        # supabase: Client = create_client(url, key)
-
-        # email: str = kwargs.get('email')
-        # password: str = kwargs.get('password')
-        # email = 'woodwardmw@gmail.com'
-        # password = 'password'
-        # try:
-        #     data = supabase.auth.sign_in_with_password({ "email": email, "password": password})
-        # except Exception as e:
-        #     raise PermissionException("Cannot authenticate user in Supabase")
-        # session = supabase.auth.get_session(data['access_token'])
-
-        # Until Supabase or something else is ready
-        token = kwargs.get('token')
-        if token.get_secret_value() != os.getenv('CHATBOT_ACCESS_TOKEN', "chatchatchat"):
-            raise PermissionException("Wrong access token!!!")
+        # Extract the access token from the request headers
+        access_token = kwargs.get('token')
+        if not access_token:
+            raise ValueError("Access token is missing")
+        access_token_str = access_token.get_secret_value()
+
+        # Verify the access token using Supabase secret
+        supabase_url: str = os.environ.get("SUPABASE_URL")
+        supabase_key: str = os.environ.get("SUPABASE_KEY")
+        supabase: Client = create_client(supabase_url, supabase_key)
+        try:
+            user_data = supabase.auth.get_user(access_token_str)
+        except gotrue.errors.AuthApiError as e:
+            raise PermissionException("Unauthorized access. Invalid token.") from e
+        result = supabase.table('userTypes').select('''
+                sources
+                '''
+            ).eq(
+            'user_type', user_data.user.user_metadata.get('user_type')
+            ).limit(1).single().execute()
+        label = result.data.get('sources')
+
+        if not label:
+            raise PermissionException(
+            "Unauthorized access. User does not have access to any sources."
+            )
+
+        kwargs['label'] = label
```

Comment:
As we are moving towards allowing multiple source types to be included in the same chat, in the auth also we need list of source-labels, not just one.